_source/_posts/2022-09-26-java-jetty-oauth.md

```diff
@@ -0,0 +1,1017 @@
+---
+layout: blog_post
+title: "Get Started with Jetty, Java, and OAuth"
+author: andrew-hughes
+by: contractor
+communities: [java]
+description: "Learn how to use Java and embedded Jetty to create a simple servlet-based web service. Then see how to do it with Spring Boot."
+tags: [java, jetty, oauth2]
+tweets:
+- "Jetty is a highly scalable Java web server and servlet engine. This tutorial shows how to use it with servlets and @springboot."
+- "Want to learn how to use Java and embedded @JettyProject to develop REST APIs? This tutorial is for you!"
+- "Build a Java REST API with @JettyProject and learn how to lock it down with OAuth 2.0 in this tutorial."
+image: blog/featured/okta-java-short-bottle-headphones.jpg
+type: conversion
+---
+
+Jetty is a small, highly-scalable Java-based web server and servlet engine. It supports HTTP/2, WebSockets, and many other protocols. It powers websites and frameworks, both large and small, such as Google AppEngine. Because it is an Eclipse project, its open source project is called Eclipse Jetty. It is standards-compliant and open source, as well as commercially usable. It is the main alternative to Tomcat when hosting Java applications. Like Tomcat, you can use Jetty both embedded and stand-alone.
+
+By default, Spring Boot creates applications with embedded web servers, which means that the server is embedded within the application code itself, so you don't have to run a separate web server to publish Java web applications. However, with a little configuration, you can also publish a WAR file to a separate Jetty or Tomcat servlet container (old-school application server style). Spring also uses Tomcat by default, but you can easily change this, as you'll see.
+
+In this tutorial, you will build a simple web service with Jetty embedded. After that, you will create the same web service in Spring Boot and Jetty. Finally, you'll add JWT (JSON Web Token) authentication and authorization to the web service using method-level security with Okta as the OAuth/OIDC provider.
+
+{% include toc.md %}
+
+## Install the Project Dependencies
+
+Before you start, please make sure you have the following prerequisites installed (or install them now).
+
+- [Java 11](https://adoptium.net/): or use [SDKMAN!](https://sdkman.io/) to manage and install multiple versions
+- [Okta CLI](https://cli.okta.com/manual/#installation): the Okta command-line interface
+- [HTTPie](https://httpie.org/doc#installation): a simple tool for making HTTP requests from a Bash shell
```

Comment:
```suggestion
- [HTTPie](https://httpie.org/doc#installation): a simple tool for making HTTP requests from the command line
```

I changed it to be more generic since I use Zsh.